src/main/java/iggly/duke/Storage.java

```diff
@@ -0,0 +1,87 @@
+package iggly.duke;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.util.ArrayList;
+
+import iggly.model.Task;
+
+/**
+ * The {@link Storage} class is responsible for handling the reading, writing and saving task data to a file.
+ */
+public class Storage {
+    private final File file;
+
+    /**
+     * Constructs a {@link Storage} instance with the specified file path.
+     *
+     * @param filePath The file path where task data is stored or will be stored.
+     */
+    public Storage(String filePath) {
+        this.file = new File(filePath);
+    }
+
+    /**
+     * Checks if the file associated with this storage exists.
+     *
+     * @return {@code true} if the file exists, {@code false} otherwise.
+     */
+    public boolean isFileExists() {
```

Comment:
isExistingFile( ) might be a better alterative. You're not required to change! Just according to the code quality, this order might be more "sensible" as the textbook puts it.